File: backend/app/routers/presets.py
"""Router for search parameter presets."""

from fastapi import APIRouter

from app.models import RADIUS_PRESETS, PresetResponse

router = APIRouter()


@router.get("/presets", response_model=PresetResponse)
async def get_search_presets():
    """
    Get recommended search parameters and UI labels for different place types.
    Aligns client UI with the unified search policy engine.
    """
    return PresetResponse(
        max_radius=5000,
        default_by_type=RADIUS_PRESETS,
        radius_options=[500, 1000, 1500, 2000, 2500, 3000, 5000],
        default_mode="auto",
        notes="auto selects 'around' for schools and 'bbox' for B2B by default",
        type_labels_tr={
            "factory": "Fabrika",
            "office": "Ofis",
            "workshop": "Atölye / İmalathane",
            "kindergarten": "Anaokulu",
            "primary_school": "İlkokul",
            "middle_school": "Ortaokul",
            "high_school": "Lise",
            "private_school": "Özel Okul",
            "college_keyword": "Kolej",
            "college_university": "Üniversite",
            "hotel": "Otel",
            "company": "Şirket",
            "holding": "Holding",
            "real_estate": "Emlak Ofisi",
            "language_school": "Dil Kursu",
            "travel_agency": "Seyahat Acentesi",
            "zoo_aquarium": "Hayvanat Bahçesi & Akvaryum",
            "theme_park": "Tema & Su Parkı",
            "museum": "Müze",
            "botanical_garden": "Botanik Bahçesi",
            "nature_park": "Milli Park & Doğa Alanı",
        },
        type_groups_tr=[
            {
                "group": "İşletmeler",
                "types": ["factory", "company", "holding", "office", "workshop"],
            },
            {
                "group": "Eğitim Kurumları",
                "types": [
                    "kindergarten",
                    "primary_school",
                    "middle_school",
                    "high_school",
                    "private_school",
                    "college_keyword",
                    "college_university",
                    "language_school",
                ],
            },
            {
                "group": "Konaklama & Hizmet",
                "types": ["hotel", "real_estate", "travel_agency"],
            },
            {
                "group": "Gezi & Eğlence",
                "types": [
                    "zoo_aquarium",
                    "theme_park",
                    "museum",
                    "botanical_garden",
                    "nature_park",
                ],
            },
        ],
    )

File: src/lib/presets.ts
import type { PlaceType } from "./types";

/**
 * Recommended default radius (meters) for each place type.
 */
export const DEFAULT_RADIUS_BY_TYPE: Record<PlaceType, number> = {
  factory: 5000,
  office: 3000,
  workshop: 3000,
  kindergarten: 1500,
  primary_school: 1500,
  middle_school: 1500,
  high_school: 2000,
  private_school: 5000,
  college_keyword: 5000,
  // Universiteler seyrek dagiliyor; backend RADIUS_PRESETS ile ayni deger.
  college_university: 5000,
  hotel: 3000,
  company: 3000,
  holding: 5000,
  real_estate: 2000,
  language_school: 2000,
  travel_agency: 2000,
  zoo_aquarium: 5000,
  theme_park: 5000,
  museum: 3000,
  botanical_garden: 5000,
  nature_park: 5000,
};

/**
 * Available radius options for the selector.
 */
export const RADIUS_OPTIONS = [
  { value: 500, label: "500 m" },
  { value: 1000, label: "1 km" },
  { value: 1500, label: "1.5 km" },
  { value: 2000, label: "2 km" },
  { value: 3000, label: "3 km" },
  { value: 5000, label: "5 km" },
  { value: 7500, label: "7.5 km" },
  { value: 10000, label: "10 km" },
] as const;
